docs/copilot/agents/overview.md

```diff
@@ -0,0 +1,234 @@
+
+In VS Code, agents are AI-powered assistants that can perform coding tasks with varying levels of autonomy. Some respond to a simple prompt and make targeted edits, while others can plan multi-step changes, run terminal commands, and orchestrate tools to achieve complex objectives.
+
+These autonomous agents can:
+
+* Understand complex context across your codebase
+* Break down high-level requirements into concrete coding tasks
+* Execute tasks independently, such as making code edits, running tests, or terminal commands
+* Invoke external tools and services to gather information or perform tasks
+* Read signals from your environment, like stack traces, linting errors, or debug state
+* React to feedback from your environment or tools and self-correct when encountering errors, such as test failures, merge conflicts, debug context, or terminal output
+
+Agents can run in different environments, such as locally in VS Code or in the background on your machine, or they can run remotely in the cloud.
+
+## Types of agents
+
+VS Code supports four main categories of agents, each designed for different use cases and levels of interaction:
+
+![Diagram showing agent types by environment and interaction.](./images/overview/agent-types-diagram.png)
+
+<!--
+```mermaid
+---
+config:
+    quadrantChart:
+        pointTextPadding: 10
+        pointRadius: 3
+        quadrantTextTopPadding: 15
+    themeVariables:
+        quadrant1Fill: "#1a3a5c"
+        quadrant2Fill: "#2d5a87"
+        quadrant3Fill: "#4a90c2"
+        quadrant4Fill: "#7eb8e2"
+        quadrant1TextFill: "#ffffff"
+        quadrant2TextFill: "#ffffff"
+        quadrant3TextFill: "#ffffff"
+        quadrant4TextFill: "#1a3a5c"
+        quadrantPointFill: "#ffffff"
+        quadrantPointTextFill: "#ffffff"
+---
+quadrantChart
+    title Agent types by environment and interaction
+    x-axis Local --&gt; Remote
+    y-axis Interactive --&gt; Background
+    quadrant-1 Cloud Agent
+    quadrant-2 Background Agent
+    quadrant-3 Local Agent
+    quadrant-4 "-"
+    VS Code built-in agents: [0.30, 0.30]
+    VS Code custom agents: [0.20, 0.20]
+    Copilot CLI: [0.30, 0.80]
+    "Third-party (OpenAI Codex)": [0.20, 0.70]
+    Copilot coding agent: [0.75, 0.75]
+```
+-->
+
+### Local agents
+
+**Best for**: Interactive conversations, immediate assistance, and real-time collaboration
+
+Local agents run directly within chat in VS Code and provide immediate, interactive assistance. You can create [custom agents](/docs/copilot/customization/custom-agents.md) to customize the behavior and capabilities of local agents to suit your workflow.
+
+**Key characteristics**:
+
+* Immediate response and interaction
+* Full access to your workspace context
+* Integrated with VS Code's editing experience
+
+Learn more about [using chat in VS Code](/docs/copilot/chat/copilot-chat.md).
+
+### Background agents
```

Comment:
Should there be a link in this section (and the next one) to the specific docs on cloud and background agents?